wireapp/wire-ios
# feat: resilient login when federated backend is unreachable FS-673

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-673" title="FS-673" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />FS-673</a>  [iOS] Make login resilient to backends being down
  </td></table>
  <br />
 


----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

It was not possible to login when a federated backend (from which you share a conversation) is unreachable.

### Causes

When running a slow sync and fetching the federated users from the unreachable backend, it would fail with a 533 error response. The app didn't handle this at all, so it would continuously retry the request to fetch the conversation.

### Solutions

Handle the 533 response by treating it as a permanent error, and create placeholder users.

### Dependencies

Needs releases with:

- https://github.com/wireapp/wire-ios-request-strategy/pull/522
- https://github.com/wireapp/wire-ios-transport/pull/250

### Testing

#### Test Coverage 

- Manually tested. This is a quick temporary fix so I didn't invest time in writing automated tests for this.

#### How to Test

- Be in conversation with a federated backend B.
- Login to the app.
- Verify that you see the conversation list.
- Verify that the federated users from backend B have "username unavailable" names.

### Notes 

This is a temporary fix to unblock logging in when a remote backend is down. The user can now log in, but messages can't be send conversations with users from this unreachable backend. Soon we'll address this topic properly.


----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
